src/main/java/org/opensrp/web/rest/LocationResource.java

```diff
@@ -268,4 +270,27 @@ public void setLocationService(PhysicalLocationService locationService) {
 
     }
 
+	/**
+	 * This methods provides an API endpoint that searches for a location and it's children using the provided location id
+	 * It returns the Geometry optionally if @param returnGeometry is set to true.
+	 * @param returnGeometry boolean which controls if geometry is returned
+	 * @param jurisdictionId location id
+	 * @return location together with it's children whose id matches the provided param
+	 */
+	@RequestMapping(value = "/findByIdWithChildren", method = RequestMethod.GET, produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<String> findByIdWithChildren(
+			@RequestParam(value = RETURN_GEOMETRY, defaultValue = FALSE, required = false) boolean returnGeometry,
+			@RequestParam(value = JURISDICTION_ID, required = false) String jurisdictionId) {
```

Comment:
Add a limit param of type int and pass to service to limit the number of locations returned